packages/cdi/charts/templates/cdi-cr-uninstaller.yaml

```
@@ -0,0 +1,67 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: default
+  name: cdi-cr-uninstaller
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "1"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cdi-cr-uninstaller
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "1"
+rules:
+  - apiGroups: [ "cdi.kubevirt.io" ]
+    resources: [ "cdis" ]
+    resourceNames:
+      - "cdi"
+    verbs: [ "delete" ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cdi-cr-uninstaller
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "2"
+subjects:
+  - kind: ServiceAccount
+    namespace: default
+    name: cdi-cr-uninstaller
+roleRef:
+  kind: ClusterRole
+  name: cdi-cr-uninstaller
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: default
+  name: cdi-cr-uninstaller
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+    "helm.sh/hook-weight": "3"
+spec:
+  template:
+    metadata:
+      name: cdi-cr-uninstaller
+    spec:
+      serviceAccountName: cdi-cr-uninstaller
+      restartPolicy: OnFailure
+      containers:
+        - name: cdi-cr-uninstaller
+          image: {{ .Values.uninstaller.image }}
+          command:
+            - /bin/sh
+            - -c
+            - >
+              kubectl delete cdi cdi;
```

I think this should be namespaced
```suggestion
              kubectl delete -n cdi cdi cdi;
```

---

CDI is in fact cluster-scoped currently. Namespace is not specified in the definition.

---

Indeed, you are right. I somehow missed that. Looks like I was always removing the resource using the yaml file (`kdelete -f ...`) and did not pay attention to that. Then yeah, please ignore the comment.